### ISSUE TITLE
no-construct-in-interface: only the first generic type argument is checked in eslint-plugin-awscdk

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#537]
### Describe the bug

In eslint-plugin-awscdk, `findGenericsTypeArgument` (`packages/eslint/src/core/ts-type/finder/generics-type-argument.ts`) returns only the **first** type argument (`aliasTypeArguments[0]` / `typeArguments[0]`). `no-construct-in-interface` therefore misses Construct types that appear in the second or later type argument of a generic type or tuple.

oxlint-plugin-awscdk walks all type arguments and reports these — so the two plugins diverge.

**Repro:**

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";

export interface TypeArgsProps {
  readonly control: Bucket;                       // control: reported by both
  readonly bucketList: Array<Bucket>;             // control (first type arg): reported by both
  readonly bucketsByName: Record<string, Bucket>; // only oxlint reports
  readonly bucketMap: Map<string, Bucket>;        // only oxlint reports
  readonly pair: [string, Bucket];                // only oxlint reports
}
```

### Expected Behavior

All five properties expose a `Bucket` and should be reported.

### Current Behavior

ESLint:

```
repro/no-construct-in-interface/type-args.ts   5:3  error  Interface property 'control' should not use CDK Construct type 'Bucket'. ...
repro/no-construct-in-interface/type-args.ts   7:3  error  Interface property 'bucketList' should not use CDK Construct type 'Bucket'. ...
```

Oxlint:

```
repro/no-construct-in-interface/type-args.ts:5:3: Interface property 'control' should not use CDK Construct type 'Bucket'. ...
repro/no-construct-in-interface/type-args.ts:7:3: Interface property 'bucketList' should not use CDK Construct type 'Bucket'. ...
repro/no-construct-in-interface/type-args.ts:9:3: Interface property 'bucketsByName' should not use CDK Construct type 'Bucket'. ...
repro/no-construct-in-interface/type-args.ts:11:3: Interface property 'bucketMap' should not use CDK Construct type 'Bucket'. ...
repro/no-construct-in-interface/type-args.ts:13:3: Interface property 'pair' should not use CDK Construct type 'Bucket'. ...
```

### Plugin

eslint-plugin-awscdk

### Plugin Version

4.4.0

### Language Version

TypeScript 6.0.3

### Additional Context

- Verified on `main` (35b889a), macOS, rule-isolated configs; divergence confirmed against oxlint-plugin-awscdk 1.0.0 on the same file.
- `no-construct-in-public-property-of-construct` uses the same finder (`findTypeOfCdkConstruct` → `findGenericsTypeArgument`) and should have the same blind spot (not separately runtime-verified).
- `Record<string, Bucket>` / `Map<string, Bucket>` are common shapes for passing resource collections in props.
